### PR TITLE
RR: Generating default query path based on scenario name + path validation

### PR DESCRIPTION
**Pull request by arkadius**

### CODE REVIEW

File: engine/lite/k8sDeploymentManager/src/test/scala/pl/touk/nussknacker/k8s/manager/PathUtilsTest.scala [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    val longScenarioNameWithoutNuInstanceIdDefaultPath = PathUtils.defaultPath(ProcessName(invalidK8sServiceName), None)
+    longScenarioNameWithoutNuInstanceIdDefaultPath should contain only 'a'
+    longScenarioNameWithoutNuInstanceIdDefaultPath should have length 63
+
+    val nuInstanceId = (1 to 10).map(_ => "b").mkString
+    val longScenarioNameWithNuInstanceIdDefaultPath = PathUtils.defaultPath(ProcessName(invalidK8sServiceName), Some(nuInstanceId))
+    ServicePreparer.serviceNameWithoutSanitization(Some(nuInstanceId), longScenarioNameWithNuInstanceIdDefaultPath) shouldEqual
+      ServicePreparer.serviceName(Some(nuInstanceId), longScenarioNameWithNuInstanceIdDefaultPath)
+  }
```
Comment (gadomsky) [inline engine/lite/k8sDeploymentManager/src/test/scala/pl/touk/nussknacker/k8s/manager/PathUtilsTest.scala:23]: * Can we have seprate test for each case? The first two are straitghforward, but the third one requires desrptive name IMO.
* Can you please also add test for ProcessName with space?
* What about special characters? I see that some of them are cut, and some replaced? Looks like bug.
```
    PathUtils.defaultPath(ProcessName("a żółć"), None) shouldEqual "a-x"
```

Comment (arkadius) [reply engine/lite/k8sDeploymentManager/src/test/scala/pl/touk/nussknacker/k8s/manager/PathUtilsTest.scala:23]: 1. 2. ok
3. It's look like group of special characters is replaced and after that is added x on the end. Take a look at K8sUtils.sanitizeName - there is plus in first regex (grouping) . I'm not sure if it is bug or intentionally added. I've rather expect one '-' for each replaced character but for me both options are fine. What do you prefer? Should I change it?

File: engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/LiteScenarioValidator.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+object LiteScenarioValidator {
+
+  def apply(config: K8sDeploymentManagerConfig) = new LiteScenarioValidator(config.nussknackerInstanceName)
+
+}
+
+case class IllegalRequestResponsePath(nussknackerInstanceName: Option[String], path: String)
+  extends Exception(s"Illegal path: $path. Path ${nussknackerInstanceName.map(i => s"after prefixation by instance name: '$i-' ").getOrElse("")}should match url path pattern and kubernetes object name pattern")
```
[the remark this answers is not in the document]
Comment (arkadius) [reply engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/LiteScenarioValidator.scala]: in scala, all Exceptions are RuntimeException. You are worry about java interops? Changed...

File: engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/PathUtils.scala [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+  private[manager] def determinePath(scenarioName: ProcessName, rrMetaData: RequestResponseMetaData) = {
+    rrMetaData.path.getOrElse(scenarioName.value)
+  }
+
+  // We don't encode url because k8s object names are more restrictively validated than urls, see https://datatracker.ietf.org/doc/html/rfc3986
+  // and all invalid characters will be clean
+  private[manager] def defaultPath(scenarioName: ProcessName, nussknackerInstanceName: Option[String]): String = {
+    val nussknackerInstancePrefixLength = nussknackerInstanceName.map(_.length + 1).getOrElse(0)
+    serviceName(None, scenarioName.value).take(63 - nussknackerInstancePrefixLength)
```
[the remark this answers is not in the document]
Comment (arkadius) [reply engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/PathUtils.scala]: Good point, extracted

File: engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/PathUtils.scala [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package pl.touk.nussknacker.k8s.manager
+
+import pl.touk.nussknacker.engine.api.RequestResponseMetaData
+import pl.touk.nussknacker.engine.api.process.ProcessName
+import pl.touk.nussknacker.k8s.manager.service.ServicePreparer.serviceName
+
+object PathUtils {
```
[the remark this answers is not in the document]
Comment (arkadius) [reply engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/PathUtils.scala]: I've added prefix - thx

File: engine/lite/k8sDeploymentManager/src/test/scala/pl/touk/nussknacker/k8s/manager/PathUtilsTest.scala (hunk 1 above)
Comment (gadomsky) [reply engine/lite/k8sDeploymentManager/src/test/scala/pl/touk/nussknacker/k8s/manager/PathUtilsTest.scala]: 3. I was also expecting that each character is replaced separatly, but your right - group replacement is also just fine.

**Review by gadomsky — APPROVED** [review]
LGTM

**Pull request merged**